pilot/pkg/model/push_context.go

```diff
@@ -349,6 +349,8 @@ func (ps *PushContext) initVirtualServices(env *Environment) error {
 	ps.VirtualServiceConfigs = vservices
 	// convert all shortnames in virtual services into FQDNs
 	for _, r := range ps.VirtualServiceConfigs {
+		r.Mutex = &sync.RWMutex{}
```

Comment:
and here. I am sure this is not right.

Reply:
This is mutex 2, why you say they are the same. And how can two different mutexes protect the same critical section?